pkg/apis/work/v1alpha1/binding_types.go

```diff
@@ -66,6 +66,33 @@ type ObjectReference struct {
 	ResourceVersion string `json:"resourceVersion,omitempty"`
 }
 
+// ReplicaRequirements represents the requirements required by each replica.
+type ReplicaRequirements struct {
+	// NodeClaim represents the node claim NodeAffinity, NodeSelector and Tolerations required by each replica.
+	// +optional
+	NodeClaim *NodeClaim `json:"nodeClaim,omitempty"`
+
+	// ResourceRequest represents the resources required by each replica.
+	// +optional
+	ResourceRequest corev1.ResourceList `json:"resourceRequest,omitempty"`
+}
+
+// NodeClaim represents the node claim NodeAffinity, NodeSelector and Tolerations required by each replica.
+type NodeClaim struct {
+	// A node selector represents the union of the results of one or more label queries
+	// over a set of nodes; that is, it represents the OR of the selectors represented
+	// by the node selector terms.
+	// +optional
+	NodeAffinity *corev1.NodeSelector `json:"nodeAffinity,omitempty" protobuf:"bytes,1,opt,name=nodeAffinity"`
```

Comment:
Better clarify that only `RequiredDuringSchedulingIgnoredDuringExecution` of `Pod.Affinity.NodeAffinity` is included here.
Otherwise it's confusing to users that why we use a different type here, comparing to `Pod.Affinity.NodeAffinity` of upstream Kubernetes.

Reply:
ACK, good advice.